buffer/src/main/java/io/netty/buffer/UnpooledDirectByteBuf.java

```diff
@@ -491,21 +491,7 @@ private void getBytes(int index, OutputStream out, int length, boolean internal)
         if (length == 0) {
             return;
         }
-
-        if (buffer.hasArray()) {
-            out.write(buffer.array(), index + buffer.arrayOffset(), length);
-        } else {
-            byte[] tmp = new byte[length];
-            ByteBuffer tmpBuf;
-            if (internal) {
-                tmpBuf = internalNioBuffer();
-            } else {
-                tmpBuf = buffer.duplicate();
-            }
-            tmpBuf.clear().position(index);
-            tmpBuf.get(tmp);
-            out.write(tmp);
-        }
+        ByteBufUtil.readBytes(internal ? internalNioBuffer() : buffer.duplicate(), index, length, out);
```

Comment:
when I tried to improve this I was wondering if we can avoid that duplicate call?

Reply:
@CodingFabian nope we can't... `internal` will be false if you use `get*`. This is needed as it may be called from multiple threads at the same time which should be fine as you not expect to do any internal updates on the indices. If you use a `get*` method there will be no duplicate invoked